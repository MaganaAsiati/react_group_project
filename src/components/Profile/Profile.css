.profile-container {
  display: flex;
  justify-content: center;
  margin: 50px 15px;
}

.profile {
  display: grid;
  grid-template-columns: 1fr;
  gap: 30px;
  width: 100%;
  max-width: 1000px;
  padding: 1.5rem 2rem;
  outline: 1px solid var(--color-grey);
  border-radius: 15px;
  box-shadow: 0 5px 10px var(--color-grey-light);
}

.profile h2 {
  margin: 0;
  margin-bottom: 20px;
  letter-spacing: 0.07rem;
  font-size: 18px;
}

.profile hr {
  height: 0;
  border: none;
  border-bottom: 1px solid var(--color-grey);
  width: 100%;
  margin-bottom: 25px;
}

.profile-sub-section {
  display: flex;
  flex-direction: column;
}

.profile-item {
  margin-bottom: 12px;
  padding: 1.2rem 1rem;
  border-radius: 8px;
  outline: 1px solid var(--color-grey);
  box-shadow: 0 5px 10px var(--color-grey-light);
  font-weight: 500;
}

@media screen and (min-width: 768px) {
  .profile {
    grid-template-columns: 1fr 1fr;
  }
}
