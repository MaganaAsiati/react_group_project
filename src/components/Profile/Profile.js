import React from 'react';
import { useSelector } from 'react-redux';
import './Profile.css';
import ProfileMissions from './profile_components/ProfileMissions';
import ProfileRocket from './profile_components/ProfileRocket';

export default function Profile() {
  const missionData = useSelector((state) => state.missionsReducer);
  const rocketData = useSelector((state) => state.rockets);

  return (
    <div className="profile-container">
      <div className="profile">
        <div className="profile-sub-section profile-rockets">
          <h2>My Rockets</h2>
          <hr />
          {rocketData.filter((data) => data.reserved)
            .map((filteredData) => (
              <ProfileRocket
                key={filteredData.id}
                name={filteredData.name}
              />
            ))}
        </div>
        <div className="profile-sub-section profile-missions">
          <h2>My Missions</h2>
          <hr />
          {missionData.filter((data) => data.reserved)
            .map((filteredData) => (
              <ProfileMissions
                key={filteredData.mission_id}
                missionName={filteredData.mission_name}
              />
            ))}
        </div>
      </div>
    </div>
  );
}
