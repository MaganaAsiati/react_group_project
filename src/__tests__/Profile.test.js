import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../Redux/configureStore';
import Profile from '../components/Profile/Profile';

describe('Profile tests', () => {
  it('render test', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
