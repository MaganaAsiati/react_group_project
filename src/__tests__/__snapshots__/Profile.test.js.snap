// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile tests render test 1`] = `
<div
  className="profile-container"
>
  <div
    className="profile"
  >
    <div
      className="profile-sub-section profile-rockets"
    >
      <h2>
        My Rockets
      </h2>
      <hr />
    </div>
    <div
      className="profile-sub-section profile-missions"
    >
      <h2>
        My Missions
      </h2>
      <hr />
    </div>
  </div>
</div>
`;
